### PR TITLE
Prefix remaining CMake options except for CMake BUILD* variables

### PR DESCRIPTION
**Pull request by byrnHDF**
Renamed remaining HDF5 library CMake options except for CMake BUILD* variables
    
      DEFAULT_API_VERSION to HDF5_DEFAULT_API_VERSION
      DISABLE_PDB_FILES to HDF5_DISABLE_PDB_FILES
      ONLY_SHARED_LIBS to HDF5_ONLY_SHARED_LIBS
      ALLOW_UNSUPPORTED to HDF5_ALLOW_UNSUPPORTED
      TEST_SHELL_SCRIPTS to HDF5_TEST_SHELL_SCRIPTS

      All other HDF5 library CMake options are prefixed with HDF5_

### CODE REVIEW

**Comment by byrnHDF** [comment]
Closes #4982

**Review by derobins — APPROVED** [review]

File: src/H5build_settings.autotools.c.in [hunk 1]
```diff
@@ -96,7 +96,7 @@ const char H5build_settings[]=
     "                           Threads: @THREADS@\n"
     "                      Threadsafety: @THREADSAFE@\n"
     "               Default API mapping: @DEFAULT_API_VERSION@\n"
-    "    With deprecated public symbols: @DEPRECATED_SYMBOLS@\n"
+    "    With deprecated public symbols:HDF5_DEFAULT_API_VERSIONS@\n"
```
Comment (derobins) [inline src/H5build_settings.autotools.c.in]: The Autotools symblol needs to stay the same

File: src/libhdf5.settings.autotools.in [hunk 2]
```diff
@@ -78,7 +78,7 @@ Dimension scales w/ new references: @DIMENSION_SCALES_WITH_NEW_REF@
                            Threads: @THREADS@
                       Threadsafety: @THREADSAFE@
                Default API mapping: @DEFAULT_API_VERSION@
-    With deprecated public symbols: @DEPRECATED_SYMBOLS@
+    With deprecated public symbols:HDF5_DEFAULT_API_VERSIONS@
```
Comment (derobins) [inline src/libhdf5.settings.autotools.in]: Also needs to be reverted

**Review by derobins — CHANGES_REQUESTED** [review]
Need to revert Autotools files

**Comment by byrnHDF** [comment]
> Need to revert Autotools files

Completed

**Review by derobins — APPROVED** [review]

**Pull request merged**